Separate the implementation of REST api and Websocket Realtime API.

luciofm opened the pull request:
Simplify `RocketChatClient` and delegate to `RestImpl` and `WebsocketImpl`

luciofm committed:
Separate the implementation of REST api and Websocket Realtime API.

luciofm committed:
Fix tests.

luciofm committed:
Parse RocketChatApiException for REST calls and translations support

luciofm committed:
Added gradle wrapper

luciofm committed:
Added plugin application for a run target on sample module

sacOO7 commented:
Hey this PR doesn't contain OkHTTP PR commits right? That way it will be easy for me to test

luciofm commented:
It is on top of the okhttp branch...

sacOO7 commented:
Yes, thank you :+1:

luciofm committed:
REST pinMessage.

luciofm committed:
Fix and cleanup tests

luciofm committed:
Reenable livechat module:

- Rename LiveChatAPI to LiveChatClient
- Revert Callbacks to interfaces
- Rename some LiveChat listeners do Callbacks
- Moved some classes to an internal package
- removed old LiveChat tests - need to reimplement them based on WebsocketImplTest

luciofm committed:
Remove okhttp-logging dependency

luciofm committed:
Merge pull request #31 from RocketChat/livechat

Re-enable livechat module:
